pallets/subtensor/tests/swap.rs

```diff
@@ -1048,3 +1048,296 @@ fn test_swap_total_hotkey_coldkey_stakes_this_interval_weight_update() {
         assert_eq!(weight, expected_weight);
     });
 }
+
+#[test]
+fn test_do_swap_coldkey_success() {
+    new_test_ext(1).execute_with(|| {
+        let old_coldkey = U256::from(1);
+        let new_coldkey = U256::from(2);
+        let hotkey = U256::from(3);
+        let netuid = 1u16;
+        let stake_amount = 1000u64;
+        let free_balance = 12345;
+
+        // Setup initial state
+        add_network(netuid, 13, 0);
+        register_ok_neuron(netuid, hotkey, old_coldkey, 0);
+        SubtensorModule::add_balance_to_coldkey_account(&old_coldkey, stake_amount + free_balance);
+
+        // Add stake to the neuron
+        assert_ok!(SubtensorModule::add_stake(
+            <<Test as Config>::RuntimeOrigin>::signed(old_coldkey),
+            hotkey,
+            stake_amount
+        ));
+
+        log::info!(
+            "TotalColdkeyStake::<Test>::get(old_coldkey): {:?}",
+            TotalColdkeyStake::<Test>::get(old_coldkey)
+        );
+        log::info!(
+            "Stake::<Test>::get(old_coldkey, hotkey): {:?}",
+            Stake::<Test>::get(hotkey, old_coldkey)
+        );
+
+        // Verify initial stake
+        assert_eq!(TotalColdkeyStake::<Test>::get(old_coldkey), stake_amount);
+        assert_eq!(Stake::<Test>::get(hotkey, old_coldkey), stake_amount);
+
+        assert_eq!(OwnedHotkeys::<Test>::get(old_coldkey), vec![hotkey]);
+        assert!(!OwnedHotkeys::<Test>::contains_key(new_coldkey));
+
+        // Get coldkey free balance before swap
+        let balance = SubtensorModule::get_coldkey_balance(&old_coldkey);
+        assert_eq!(balance, free_balance);
+
+        // Perform the swap
+        assert_ok!(SubtensorModule::do_swap_coldkey(
+            <<Test as Config>::RuntimeOrigin>::signed(old_coldkey),
+            &old_coldkey,
+            &new_coldkey
+        ));
+
+        // Verify the swap
+        assert_eq!(Owner::<Test>::get(hotkey), new_coldkey);
+        assert_eq!(TotalColdkeyStake::<Test>::get(new_coldkey), stake_amount);
+        assert!(!TotalColdkeyStake::<Test>::contains_key(old_coldkey));
+        assert_eq!(Stake::<Test>::get(hotkey, new_coldkey), stake_amount);
+        assert!(!Stake::<Test>::contains_key(hotkey, old_coldkey));
+        assert_eq!(OwnedHotkeys::<Test>::get(new_coldkey), vec![hotkey]);
+        assert!(!OwnedHotkeys::<Test>::contains_key(old_coldkey));
+
+        // Verify balance transfer
+        assert_eq!(SubtensorModule::get_coldkey_balance(&new_coldkey), balance);
```

Comment:
```suggestion
        assert_eq!(SubtensorModule::get_coldkey_balance(&new_coldkey), balance + balance_new_coldkey);
```
Goes with earlier comment